src/vanilla/store.ts

```diff
@@ -519,14 +519,83 @@ const buildStore = (
     }
   }
 
+  const peekAtomState = <Value>(
```

Comment:
a lot of `peekAtomState` code looks similar to `readAtomState`. Is it cleaner to have this separate function or to add more conditionals to `readAtomState`?